### PR TITLE
ethtool: resources/ethtool: implement ETHTOOL_GLINKSETTINGS

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

File: lxd/resources/network_ethtool.go [hunk 1]
```diff
@@ -219,6 +247,108 @@ func ethtoolGset(ethtoolFd int, req *ethtoolReq, info *api.ResourcesNetworkCardP
...
+	if errno != 0 {
+		return errors.Wrap(unix.Errno(errno), "Failed to ETHTOOL_GLINKSETTINGS")
+	}
+
+	// This insane interface gives us the size of the masks as a negative value.
+	if ethLinkSettings.linkModeMasksNwords >= 0 || ethLinkSettings.cmd != 0x0000004c {
+		return errors.Wrap(unix.Errno(unix.EINVAL), "Failed to ETHTOOL_GLINKSETTINGS")
+	}
+
+	ethLinkSettings.cmd = 0x0000004c
```
Comment (stgraber) [inline lxd/resources/network_ethtool.go:268]: Why does it need to be set again?

File: lxd/resources/network_ethtool.go [hunk 2]
```diff
@@ -219,6 +247,108 @@ func ethtoolGset(ethtoolFd int, req *ethtoolReq, info *api.ResourcesNetworkCardP
...
+	// Set the size of the maks we want to retrieve.
```
Comment (stgraber) [inline lxd/resources/network_ethtool.go:269]: s/maks/masks/

File: lxd/resources/network_ethtool.go (hunk 1 above)
Comment (brauner) [reply lxd/resources/network_ethtool.go:268]: Not sure if the kernel clobbers it when it does a `copy_to_user()`. So better safe.

File: lxd/resources/network_ethtool.go (hunk 2 above)
Comment (brauner) [reply lxd/resources/network_ethtool.go:269]: Whoops.

File: lxd/resources/network_ethtool.go (hunk 1 above)
Comment (brauner) [reply lxd/resources/network_ethtool.go:268]: Wait, it can't.

File: lxd/resources/network_ethtool.go [hunk 3]
```diff
@@ -117,6 +117,34 @@ type ethtoolValue struct {
 	data uint32
 }
 
+const EthtoolLinkModeMaskMaxKernelNu32 = 127 // SCHAR_MAX
```
Comment (stgraber) [inline lxd/resources/network_ethtool.go:120]: This should be `ethtoolLinkModeMaskMaxKernelNu32`, we don't need it exported.

**Pull request merged**